client/src/EventListener/UnauthorizedExceptionListener.php

```diff
@@ -11,11 +11,14 @@ class UnauthorizedExceptionListener
 {
     public function onKernelException(ExceptionEvent $exceptionEvent)
     {
+        $request = $exceptionEvent->getRequest();
         $exception = $exceptionEvent->getThrowable();
 
         if ($exception instanceof HttpException) {
             if (Response::HTTP_UNAUTHORIZED == $exception->getStatusCode()) {
-                $response = new RedirectResponse('/login?from=api');
+                $url = '/login?from=api&lastPage=' . urlencode($request->getRequestUri());
```

Comment:
Do we actually use the from=api anywhere? Seems like an odd construct.

Reply:
Not that I've seen. I thought it was weird, at a guess I'm assuming it's stating the error has been passed from the API

Reply:
[Used here](https://github.com/ministryofjustice/opg-digideps/blob/main/client/src/Controller/IndexController.php#L110). Convoluted bit of code though!